presto-main/src/main/java/io/prestosql/operator/scalar/ArrayIntersectFunction.java

```diff
@@ -40,6 +47,7 @@ public ArrayIntersectFunction(@TypeParameter("E") Type elementType)
     @SqlType("array(E)")
     public Block intersect(
             @TypeParameter("E") Type type,
+            @OperatorDependency(operator = IS_DISTINCT_FROM, returnType = BOOLEAN, argumentTypes = {"E", "E"}) MethodHandle elementIsDistinctFrom,
```

Comment:
Adding methodHandle dependencies in the methods that get called row per row is a bit of an anti-pattern. It typically means the code is turning around and doing `methodHandle.invoke` at some point, which become polymorphic when the function is used in a variety of contexts. A better approach is to get the dependencies during the "get implementation" call, which should assemble a methodhandle chain or generate specialized bytecode. 

It doesn't matter for this specific instance since it's hard to assemble the methodhandle chain with the TypedSet in the middle, but I'm just pointing it out as a general comment (so no need to change anything here).

Reply:
left as is